<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>923</width>
    <height>607</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="grid" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>450</width>
      <height>450</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="minimumSize">
     <size>
      <width>450</width>
      <height>450</height>
     </size>
    </property>
    <property name="baseSize">
     <size>
      <width>450</width>
      <height>450</height>
     </size>
    </property>
    <property name="autoFillBackground">
     <bool>false</bool>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: rgb(27, 27, 27);</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>100</y>
      <width>301</width>
      <height>61</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>    Mouse Movement</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="mouse_movement">
         <property name="frameShape">
          <enum>QFrame::Panel</enum>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Mouse Pressed</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="mouse_pressed">
         <property name="frameShape">
          <enum>QFrame::Panel</enum>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>50</y>
      <width>251</width>
      <height>24</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QCheckBox" name="show_axes">
       <property name="text">
        <string>Show Axes</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="show_grid">
       <property name="text">
        <string>Show grid</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>510</y>
      <width>112</width>
      <height>32</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear Grid</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="AlgoBox">
    <property name="geometry">
     <rect>
      <x>490</x>
      <y>190</y>
      <width>391</width>
      <height>251</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="Line">
     <attribute name="title">
      <string>Line</string>
     </attribute>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>50</y>
        <width>130</width>
        <height>61</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QComboBox" name="line_algorithm">
         <property name="editable">
          <bool>false</bool>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>DDA</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Bresenham</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="Draw">
         <property name="text">
          <string>Draw Line</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>50</y>
        <width>51</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>DDA  :</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>90</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Bresenham :</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>20</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
        <weight>75</weight>
        <bold>true</bold>
        <underline>true</underline>
       </font>
      </property>
      <property name="text">
       <string>Time Taken</string>
      </property>
     </widget>
     <widget class="QLabel" name="DDA_time">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>50</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLabel" name="Bresenham_time">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>85</y>
        <width>91</width>
        <height>21</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Clipping</string>
     </attribute>
     <widget class="QPushButton" name="Transformation_polygon">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>80</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Start Polygon</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="width">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>30</y>
        <width>46</width>
        <height>25</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QSpinBox" name="height">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>30</y>
        <width>46</width>
        <height>25</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>30</y>
        <width>31</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>X :</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>30</y>
        <width>31</width>
        <height>21</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
        <weight>75</weight>
        <bold>true</bold>
       </font>
      </property>
      <property name="text">
       <string>Y :</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Line_Clipping">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>30</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Line Clipping</string>
      </property>
     </widget>
     <widget class="QPushButton" name="Polygon_Clipping">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>80</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Polygon Clipping</string>
      </property>
     </widget>
     <widget class="QPushButton" name="draw_rectangle">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>131</width>
        <height>31</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>9</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Draw Rectangle</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>50</y>
      <width>128</width>
      <height>32</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="gridsize_label">
       <property name="text">
        <string>Grid Size</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="gridsize">
       <property name="minimum">
        <number>1</number>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>923</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>grid</class>
   <extends>QWidget</extends>
   <header>grid.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
